controllers/v1/bulkUploadRequest.js

```diff
@@ -5,159 +5,369 @@
-     * @apiGroup Bulk Upload
-     * @apiHeader {String} X-authenticated-user-token Authenticity token
-     * @apiSampleRequest /admin-service/api/v1/bulkUploadRequest/details
-     * @apiUse successBody
-     * @apiUse errorBodyuser
-    */
+  /**
+   * @api {get} /admin-service/api/v1/bulkUploadRequest/getDownloadableUrls/_id
+   * Get downloadable url of bulk csv files
+   * @apiVersion 1.0.0
+   * @apiGroup Bulk Upload
+   * @apiHeader {String} X-authenticated-user-token Authenticity token
+   * @apiSampleRequest /admin-service/api/v1/bulkUploadRequest/getDownloadableUrls/p0f4n3o1kamjmacu?fileType=input
+   * @apiUse successBody
+   * @apiUse errorBodyuser
+   * @apiParamExample {json} Response:
+   * {
+   * "message": "Url's generated successfully",
+   * "status": 200,
+   * "result": {
+   *     "filePath": "d04c5432-cb7e-4bbe-ace9-1df412117ae5/1590414313_52082.csv",
+   *     "url": "https://storage.googleapis.com/download/storage/v1/b/sl-dev-storage/o/d04c5432-cb7e-4bbe-ace9-1df412117ae5%2F1590414313_52082.csv?generation=1590414313701670&alt=media"
+   *  }
+   * }
+  */
 
   /**
-   * to get details of the upload request
+   * to get getDownloadable url
    * @method
    * @name details
    * @param  {req}  - requested data.
-   * @returns {json} Response consists of request details
+   * @returns {json} Response consists of getDownloadable Url
+  */
+
+  getDownloadableUrls(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let response = await bulkUploadHelper.getDownloadableUrls(req.userDetails.userToken, req.params._id, req.query.fileType);
+        return resolve(response);
+
+      } catch (error) {
+
+        return reject({
+          status:
+            error.status ||
+            httpStatusCode["internal_server_error"].status,
+
+          message:
+            error.message ||
+            httpStatusCode["internal_server_error"].message
+        });
+      }
+    });
+  }
+
+
+  /**
+  * @api {get} /admin-service/api/v1/bulkUploadRequest/getStatus 
+  * Get all status of bulk upload
+  * @apiVersion 1.0.0
+  * @apiGroup Bulk Upload
+  * @apiHeader {String} X-authenticated-user-token Authenticity token
+  * @apiSampleRequest /admin-service/api/v1/bulkUploadRequest/getStatus
+  * @apiUse successBody
+  * @apiUse errorBodyuser
+  * @apiParamExample {json} Response:
+  {
+   "message": "Status list featched successfully",
+   "status": 200,
+   "result": [
+       {
+           "label": "All",
+           "value": "all"
+       },
+       {
+           "label": "Completed",
+           "value": "completed"
+       },
+       {
+           "label": "Proccessing",
+           "value": "proccessing"
+       }
+   ]
+}
+**/
+
+  /**
+   * To get all status
+   * @method
+   * @name getStatus
+   * @param  {req}  - requested data.
+   * @returns {json} Response consists of status list
   */
 
- details(req) {
-  return new Promise(async (resolve, reject) => {
-    try {
+  getStatus(req) {
```

Comment:
@rakeshSgr - Syggest naming it to statusList

Reply:
resolved